dialogue-clients/src/main/java/com/palantir/dialogue/clients/ReloadingClientFactory.java

```diff
@@ -531,6 +534,13 @@ private ImmutableList<TargetUri> getTargetUris(
                 }
             }
         }
+        if (targetUris.isEmpty() && failedToParse) {
+            // Handle cases like "host:-1", but only when _all_ uris are invalid
+            log.warn(
+                    "Failed to parse all URIs, falling back to legacy DNS approach for service '{}'",
+                    SafeArg.of("service", serviceNameForLogging));
+            return uris.stream().map(TargetUri::of).collect(ImmutableList.toImmutableList());
```

Comment:
Later on we parse uri strings into the `URL` type. I wrote an implementation of this fix which uses URL throughout, however that has other issues (requires known schemes, so it would throw if `mesh-http://` uris are read in places we don't test). I think this is the safest path forward.